selenium/spec/asset_library_urls_spec.rb

```diff
@@ -0,0 +1,123 @@
+    @cal_net= CalNetPage.new @driver
+    @cal_net.log_in(WebDriverUtils.admin_username, WebDriverUtils.admin_password)
+    @course_id = @canvas.create_complete_test_course(test_id, test_users)
+    @canvas.log_out
+    @cal_net.logout_success_message_element.when_visible WebDriverUtils.page_load_wait
+    @canvas.load_homepage
+    @cal_net.log_in(test_teacher['username'], WebDriverUtils.test_user_password)
+    @canvas.load_course_site @course_id
+    @canvas.accept_login_messages @course_id
+    @asset_library = AssetLibraryPage.new @driver
+    @asset_library_url = @canvas.click_asset_library_link
+    @asset_library.load_page(@driver, @asset_library_url)
+    @asset_library.click_manage_categories_link
+    @asset_library.add_category 'Category 1'
+    @asset_library.add_category 'Category 2'
+    @asset_library.delete_category(@driver, 1)
+  end
+
+  before(:example) do
+    @asset_library.load_page(@driver, @asset_library_url)
+    @asset_library.click_add_site_link
+  end
+
+  it 'allows the user to add a URL, title, category, and description to the asset library' do
+    @asset_library.enter_url_metadata('en.wikipedia.org', 'URL Title 1', 'Category 1', 'URL description')
+    @asset_library.click_add_url_button
+    @asset_library.wait_for_asset_in_gallery(@driver, 'URL Title 1')
+    expect(@asset_library.gallery_asset_owner_name_elements[0].text).to eql("by #{test_teacher['fullName']}")
+    @asset_library.click_asset_link 0
+    @asset_library.wait_for_asset_detail(@driver, 'URL Title 1')
+    # TODO: verify asset category, description, and embedded URL once visible in the viewport
+  end
+
+  it 'allows the user to add a URL, title, and category to the asset library' do
+    @asset_library.enter_url_metadata('en.wikipedia.org', 'URL Title 2', 'Category 1', nil)
+    @asset_library.click_add_url_button
+    @asset_library.wait_for_asset_in_gallery(@driver, 'URL Title 2')
+    expect(@asset_library.gallery_asset_owner_name_elements[0].text).to eql("by #{test_teacher['fullName']}")
+    @asset_library.click_asset_link 0
+    @asset_library.wait_for_asset_detail(@driver, 'URL Title 2')
+    # TODO: verify asset category and embedded URL once visible in the viewport
+  end
+
+  it 'allows the user to add a URL and title to the asset library' do
+    @asset_library.enter_url_metadata('en.wikipedia.org', 'URL Title 3', nil, 'URL description')
+    @asset_library.click_add_url_button
+    @asset_library.wait_for_asset_in_gallery(@driver, 'URL Title 3')
+    expect(@asset_library.gallery_asset_owner_name_elements[0].text).to eql("by #{test_teacher['fullName']}")
+    @asset_library.click_asset_link 0
+    @asset_library.wait_for_asset_detail(@driver, 'URL Title 3')
+    # TODO: verify asset embedded URL once visible in the viewport
+  end
+
+  it 'allows the user to add a URL, title, and description to the asset library' do
+    @asset_library.enter_url_metadata('en.wikipedia.org', 'URL Title 4', nil, 'URL description')
```

Comment:
@nicolaasmatthijs from a design perspective, I like to be as explicit as possible about workflows in the tests themselves ... and therefore conservative about combining operations on the page.  But I agree that this is a pretty good candidate for consolidation.